### ISSUE TITLE
Feature: Add options to selectively disable certain automatic dependency tracking.

### ISSUE THREAD

**Issue by ctolkien** [microsoft/ApplicationInsights-dotnet#2838]
We have scenarios where we would like to track outbound http requests, but not track SQL. There does not seem to be a way to selectively enable only HTTP or SQL, we must disable the entire module.

**Comment by cijothomas** [comment]
That is true! The only workaround is to use TelemetryProcessor to selectively drop unwanted telemetry.

**Comment by zhimin-z** [comment]
This issue is stale because it has been open 300 days with no activity. Remove stale label or this will be closed in 7 days. Commenting will instruct the bot to automatically remove the label.

**Comment by ctolkien** [comment]
I'd like this kept open. This is a consistent pain point for our usage.

**Comment by rajkumar-rangaraj** [comment]
There are currently no plans to add an option to control this behavior. You may need to create a custom OpenTelemetry activity processor using version 3.x to achieve this functionality.

https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-filter?source=recommendations&tabs=aspnetcore
